lib/widgets/app.dart

```diff
@@ -129,7 +135,11 @@ class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
                 (_) => widget._declareReady());
             }
             GlobalLocalizations.zulipLocalizations = ZulipLocalizations.of(context);
-            return child!;
+            final locale = Localizations.localeOf(context); // Get locale from context
+            return Directionality(
+              textDirection: locale.languageCode == 'ar' ? TextDirection.rtl : TextDirection.ltr,
```

Comment:
Huh, is this needed? I would have thought that Flutter would supply this automatically, in `MaterialApp` (and `WidgetsApp` which it's built around).

Looking in the Flutter source, it sure does look that way. It's also mentioned in the docs of [`Localization`](https://main-api.flutter.dev/flutter/widgets/Localizations-class.html), which is one of the things [`WidgetsApp`](https://main-api.flutter.dev/flutter/widgets/WidgetsApp-class.html) provides.

Suppose you leave this change out; how does the behavior change?

(If we do need a version of this change, it'll be another example of a change that should be its own separate commit.)